### DISCUSSION TITLE
pymupdf use on private data?

### DISCUSSION THREAD

**Discussion by nishant0001** [pymupdf/PyMuPDF#2555] [Q&A]
@julian-smith-artifex-com @JorjMcKie @rk700 @jamie-lemon
please tell me using pymupdf on secured privacy pdf is safe or not?
is there any vulnerability?

**Comment by JorjMcKie** [comment]
I am still underway, so maybe I can answer with more verbosity tomorrow.
But PyMuPDF fully supports PDF encryption for both, input and output.
 It also does not access any resources beyond the process within which it is running. Specifically, it doesn't access internet resources.
PyMuPDF is open source as well as its base library MuPDF.
Otherwise, PyMuPDF does not add any exposure, so the ultimate answer would be given by MuPDF.
